### PR TITLE
Fix some lint errors

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

File: apis/metal3.io/v1alpha1/baremetalhost_types.go [hunk 1]
```diff
@@ -118,6 +118,8 @@ const (
 	// has any sort of error.
 	OperationalStatusError OperationalStatus = "error"
 
+	// OperationalStatusDelayed is the status value for when the host
+	// deployment needs to be delayed to limit simlultaneous hosts provisioning
```
Comment (furkatgofurov7) [inline apis/metal3.io/v1alpha1/baremetalhost_types.go:122]: tiny nit: 'simultaneous'

**Comment by dhellmann** [comment]
These all look OK. I wonder how the linter job we have didn't catch them. Do we need to change some settings?

File: apis/metal3.io/v1alpha1/baremetalhost_types.go (hunk 1 above)
Comment (elfosardo) [reply apis/metal3.io/v1alpha1/baremetalhost_types.go:122]: d'oh! Me and my typos :/

**Comment by elfosardo** [comment]
> These all look OK. I wonder how the linter job we have didn't catch them. Do we need to change some settings?

apparently we do, most of the linters in golangci-lint are disabled by default
I'll submit a PR with some basic options, we can choose if we want something more advanced later

File: pkg/provisioner/ironic/ironic.go [hunk 2]
```diff
@@ -72,6 +72,7 @@ func (e macAddressConflictError) Error() string {
 	return fmt.Sprintf("MAC address %s conflicts with existing node %s", e.Address, e.ExistingNode)
 }
 
+// NewMacAddressConflictError is a wrap for macAddressConflictError error
```
Comment (dtantsur) [inline pkg/provisioner/ironic/ironic.go:75]: This doesn't seem very informative, maybe we should improve it further.

**Review by dtantsur — COMMENTED** [review]
/lgtm

**Pull request merged**